src/index.js

```diff
@@ -1,182 +1,194 @@
+  // const dataSet = dicomParser.parseDicom(dicomp10stream)
 
-    if( !studyInstanceUid ) return;
+  const studyInstanceUid = dataSet.string("x0020000d");
 
-    // Extract uids
-    const id = callback.uids({
-        studyInstanceUid,
-        seriesInstanceUid: dataSet.string('x0020000e'),
-        sopInstanceUid: dataSet.string('x00080018'),
-        transferSyntaxUid: dataSet.string('x00020010')
-    })
+  if (!studyInstanceUid) return;
 
-    let bulkDataIndex = 0
-    let imageFrameIndex = 0
-    const generator = {
-        bulkdata: async (bulkData) =>
-            await callback.bulkdata(id, bulkDataIndex++, bulkData),
-        imageFrame: async (imageFrame) =>
-            await callback.imageFrame(id, imageFrameIndex++, imageFrame),
-        videoWriter: async (dataSet) =>
-            await callback.videoWriter(id,dataSet),
-    }
+  // Extract uids
+  const id = callback.uids({
+    studyInstanceUid,
+    seriesInstanceUid: dataSet.string("x0020000e"),
+    sopInstanceUid: dataSet.string("x00080018"),
+    transferSyntaxUid: dataSet.string("x00020010"),
+  });
 
-    // convert to DICOMweb MetaData and BulkData
-    const result = await getDataSet(dataSet, generator, options);
+  const targetId = transcodeId(id, options);
 
-    await callback.metadata(id, result.metadata)
+  let bulkDataIndex = 0;
+  let imageFrameIndex = 0;
+  const generator = {
+    bulkdata: async (bulkData) => callback.bulkdata(targetId, bulkDataIndex++, bulkData),
+    imageFrame: async (originalImageFrame) => {
+       const { imageFrame, id: transcodedId } = await transcodeImageFrame(
+        id,
+        targetId,
+        originalImageFrame,
+        dataSet,
+        options
+      );
 
-    // resolve promise with statistics
-    return {}
-}
+      return callback.imageFrame(transcodedId, imageFrameIndex++, imageFrame);
+    },
+    videoWriter: async (_dataSet) => callback.videoWriter(id, _dataSet)
+  };
 
+  // convert to DICOMweb MetaData and BulkData
+  const result = await getDataSet(dataSet, generator, options);
 
+  const transcodedMeta = transcodeMetadata(result.metadata, id, options);
 
-const OverallStats = new Stats('OverallStats', 'Overall statistics');
-const StudyStats = new Stats('StudyStats','Study Generation', OverallStats);
+  await callback.metadata(targetId, transcodedMeta);
 
-const handleHomeRelative = dirName => dirName[0]=='~' ? (path.join(homedir,dirName.substring(1))) : dirName;
+  // resolve promise with statistics
+  return {};
+};
 
-/**
- * The mkdicomweb command first runs mkdicomwebinstances, writing out the deduplicated data, and then runs the
- * mkdicomwebstudy command, creating the deduplicated data set.  This version, however, keeps the deduplicated
- * data in memory by default on a study level, which avoids needing to run the load process.
- */
-const main = async defaults => {
-    const directoryName = handleHomeRelative(getArg('-d', '--directory', dicomwebDefaultDir, 'Set output directory (~/dicomweb)'))
```

Comment:
move this entire block of getting command args/options to a specific file /program which will make use of a third party lib